prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -0,0 +1,81 @@
+package gov.cdc.prime.router.history.azure
+
+import com.microsoft.azure.functions.HttpMethod
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import com.microsoft.azure.functions.annotation.AuthorizationLevel
+import com.microsoft.azure.functions.annotation.BindingName
+import com.microsoft.azure.functions.annotation.FunctionName
+import com.microsoft.azure.functions.annotation.HttpTrigger
+import gov.cdc.prime.router.azure.HttpUtilities
+import gov.cdc.prime.router.azure.WorkflowEngine
+import gov.cdc.prime.router.tokens.AuthenticationStrategy
+import gov.cdc.prime.router.tokens.authenticationFailure
+import gov.cdc.prime.router.tokens.authorizationFailure
+
+/**
+ * Deliveries API
+ * Returns a list of Actions from `public.action`.
+ */
+class DeliveryFunction(
```

Comment:
SubmissionFunction and DeliveryFunction are almost identical. Could it just be a HistoryDataFunction (or some other good name) that creates an instance of the correct ReportFileClass? This means you would dynamically create the correct ReportFileAccess as well.

Is there a reason to have these separated out?